### DISCUSSION TITLE
on became hovered event?

### DISCUSSION THREAD

**Discussion by malted** [replit/kaboom#561] [Q&A] [answered]
Is there/will there be a way to fire an event on the frame an object becomes hovered?
i.e. the cursor was not on it last frame, and is on it this frame.

At the moment, the only way I can see to achieve this is to store the hover state of an object, then check it on every onHover tick to see if it is false. An inbuilt method would be nice.

**Comment by lajbel** [comment]
You can use onHover second function

```js
obj.onHover(
// when hover
() => {
    obj.use(area());
},
// when no hover
() => {
    obj.unuse(area())
}
);
```

**Comment by lajbel** [comment, marked as answer]
Changed the functionality of `onHover` to runs **once** when object is hovered and `onHoverEnd` to runs **once** when a object is unhovered

You still can use `onHoverUpdate()` for old behavior
